### ISSUE TITLE
[feature request] install AIR SDK through Homebrew

### ISSUE THREAD

**Issue by myflashlab** [Gamua/Adobe-Runtime-Support#425]
It would be super cool if we could install the AIR SDK through Homebrew on Mac at least (officially supported by Harman). It would help doing some magics with GitHub Actions to maybe automate some ANE developments on the fly.

And even some more magics to compile ipa and apk through GitHub Actions...

If this could happen, maybe it would be even nicer if the license file for the AIR SDK can be set through the -adt command instead of actually placing it inside the SDK folders. I’m not sure how possible it would be, I’m just thinking out loud ^_^

**Comment by zwetan** [comment]
some comments

  - on macOS
     you'll have dev who wants [homebrew](https://brew.sh/), other dev who want [macports](https://www.macports.org/)
     ideally you would want to release a `.pkg` for those reasons
     - natural native way to do it on macOS
     - can validate the use of a license
     - use case: distribution of .NET core SDK see [the build for osx](https://github.com/dotnet/runtime/tree/master/src/installer/pkg/packaging/osx)

  - on Windows
     kind of the same problem, some dev will want [chocolatey](https://chocolatey.org/), other dev will want [scoop](https://scoop.sh/), other dev will want [nuget](https://www.nuget.org/), etc.
    - Microsoft announced winget as official WIndows Package Manager
      see [Use the winget tool to install and manage applications](https://docs.microsoft.com/en-us/windows/package-manager/winget/)
     and open source project [winget-cli](https://github.com/microsoft/winget-cli/)
     problem is winget is pretty new and not wildly used
     and "The client requires Windows 10 1709 (build 16299) or later at this time."
    - usually dev distribute SDK either as targz/zip or msi/exe installers
       use case: distribution of .NET core SDK see [dotnet/installer](https://github.com/dotnet/installer)
       which is "overkill" but at least show everything that can be done etc.
    - a simple exe/msi installer would allow to "validate a license"
      for ex [Visual Studio Code on Windows - Common questions](https://code.visualstudio.com/docs/setup/windows#_common-questions)
      > What command-line arguments are supported by the Windows Setup?
      > VS Code uses [Inno Setup](https://www.jrsoftware.org/isinfo.php) to create its setup package for Windows. Thus, all the [Inno Setup command-line switches](https://www.jrsoftware.org/ishelp/index.php?topic=setupcmdline) are available for use.


Ideally just distributing a `.zip` or `.dmg` etc. would be enough but those do not enforce license usage which I guess can be an issue with the distribution of the AIR SDK, so then the comments above.

Distributing an SDK cross-platform can be complicated, depending how far you want to go ^_^

Validating the license via a command-line option like suggest @myflashlab could help imho
but not sure how practicable this is

**Comment by marchbold** [comment]
Agreed, there are too many installation managers to support them all.

I have suggested the command line input for the license to Andrew in the past, definitely think this is a good idea.  Will separate that out into a separate issue request.

**Comment by ajwfrost** [comment]
We had a request very early on to provide the SDK via HomeBrew .. the issue is that our legal team want to ensure people have read and agreed to the license agreement, especially relevant for people who might be in countries that the US have on their export control lists (Syria, Libya I think, North Korea etc) where the software is not allowed. There was a challenge I believe to being able to ensure that this is accepted by the user prior to download...

My preference would be to have an 'SDK Manager' which is a simple/free app without restrictions, that would then (a) ask the user to confirm the license agreement and then (b) download or update the SDK and take care of any installation and merging of files etc (avoiding the issues we get when Xcode formats change from files to folders like the recent .xib fiasco..). I even started trying to put together a quick Flex-based AIR app for this, back in the day when I had a bit of spare time.. a command-line version would be fine, and there's no real reason for this to be driven by us..! Community-driven and/or open source would be better...
